File: leetcode/771. Jewels and Stones/main.c
// Solved by @kitanoyoru

int numJewelsInStones(char * jewels, char * stones){
  int i, j;
  int ans = 0;

  for (i = 0; i < strlen(jewels); i++) {
    for (j = 0; j < strlen(stones); j++) {
      if (jewels[i] == stones[j]) {
        ans++;
      }
    }
  }

  return ans;
}

File: leetcode/771. Jewels and Stones/main.go
// Solved by @kitanoyoru

package main

func numJewelsInStones(jewels string, stones string) int {
	hm := make(map[rune]int)
	for _, stone := range stones {
		hm[stone]++
	}

	ans := 0
	for _, jewel := range jewels {
		ans += hm[jewel]
	}

	return ans
}
